### PR TITLE
Updated .gitignore to modern set

### PR DESCRIPTION
**Pull request by xperiandri**
As now a lot of stuff is created by Visual Studio, CodeRush, etc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,3 @@
-.DS_Store
```
Comment (xperiandri) [inline .gitignore]: Accidentally. I will restore them

File: .gitignore [hunk 2]
```diff
@@ -236,6 +245,24 @@ _Pvt_Extensions
...
+.idea/
+*.sln.iml
+
+# CodeRush
+.cr/
+
+# Python Tools for Visual Studio (PTVS)
+__pycache__/
+*.pyc
+/src/Api/global.json
```
Comment (xperiandri) [inline .gitignore]: OK

**Comment by xperiandri** [comment]
Fixed

**Comment by praeclarum** [comment]
Thank you!

**Pull request merged**